src/components/PostDetails.tsx

```diff
@@ -2,106 +1,147 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { addComment, deleteComment, getComments } from '../utils/fetchClient';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+  showForm: boolean;
+  setShowForm: (value: boolean) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  showForm,
+  setShowForm = () => {},
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+
+  const [buttonLoader, setButtonLoader] = useState(false);
+  const [commentLoader, setCommentLoader] = useState(false);
+  const [commentError, setCommentError] = useState(false);
+
+  useEffect(() => {
+    if (selectedPost) {
+      setCommentLoader(true);
+
+      getComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setCommentError(true))
+        .finally(() => setCommentLoader(false));
+    }
+  }, [selectedPost, setComments]);
+
+  const addFunction = ({ name, email, body }: CommentData) => {
+    setButtonLoader(true);
+
+    const postId = selectedPost?.id || 0;
+
+    return addComment({ postId, name, email, body })
+      .then(newComment => setComments([...comments, newComment]))
+      .catch(() => {
+        setCommentError(true);
+      })
+      .finally(() => setButtonLoader(false));
+  };
+
+  const deleteFunction = (commentId: number) => {
+    setComments(currentCommenst => {
+      return currentCommenst.filter(comment => comment.id !== commentId);
+    });
+
+    deleteComment(commentId)
+      .then(() => {})
+      .catch(error => {
+        throw error;
```

Comment:
Throwing an error directly in the `catch` block is not a good practice as it can lead to unhandled promise rejections. Consider handling the error more gracefully, perhaps by setting an error state or displaying a notification to the user.